### PR TITLE
Add skip_first_wait to profiler.schedule

### PR DESCRIPTION
**Pull request by sraikund16**
Summary:
Addresses https://github.com/pytorch/pytorch/issues/91888

We use wait as the amount you wait in between cycles when profiling and skip_first to delay the start of said profiling. However, once skip_first steps are completed, we immediately go to the wait phase. This is not problematic if wait is smaller than skip_first because we can just lower the values of skip_first, but if it is larger then we end up starting the first profile much later than desired. For example imagine a skip first of 1 and a wait of 100 with repeat of 2. We do want to wait 100 steps in between cycle 1 and 2 but we may not want to start warmup of cycle 1 at step 101 (forced because wait occurs directly after first steps skipped). This diff addresses this by adding a flag to skip the first wait.

Adds new flag but sets to false by default so that existing impl is not affected.

Test Plan:
Got reasonable traces with this schedule:

schedule=torch.profiler.schedule(
            wait=10, warmup=3, active=1, repeat=1, skip_first=1, skip_first_wait=1
        )

D66198138

@diff-train-skip-merge

### CODE REVIEW

File: torch/profiler/profiler.py [hunk 1]
```diff
@@ -434,14 +434,22 @@ class ProfilerAction(Enum):
...
+    skip_first_wait: bool = False,
 ) -> Callable:
     """
     Returns a callable that can be used as profiler ``schedule`` argument. The profiler will skip
     the first ``skip_first`` steps, then wait for ``wait`` steps, then do the warmup for the next ``warmup`` steps,
     then do the active recording for the next ``active`` steps and then repeat the cycle starting with ``wait`` steps.
     The optional number of cycles is specified with the ``repeat`` parameter, the zero value means that
-    the cycles will continue until the profiling is finished.
+    the cycles will continue until the profiling is finished. The ``skip_first_wait`` parameter controls
+    whether the first wait stage should be skipped. This can be useful if a user wants to wait larger than skip_first
```
Comment (aaronenyeshi) [inline torch/profiler/profiler.py:451]: may be typo: larger == longer

File: torch/profiler/profiler.py [hunk 2]
```diff
@@ -434,14 +434,22 @@ class ProfilerAction(Enum):
...
+    between cycles but not for the first profile.
```
Comment (aaronenyeshi) [inline torch/profiler/profiler.py:452]: would be helpful to have some example here for the wait and skip_first occurances

**Review by aaronenyeshi — APPROVED** [review]

File: torch/profiler/profiler.py (hunk 2 above)
Comment (sraikund16) [reply torch/profiler/profiler.py:452]: added example in newest diff

File: torch/profiler/profiler.py (hunk 1 above)
Comment (sraikund16) [reply torch/profiler/profiler.py:451]: grammar issue will fix

**Review by briancoutinho — APPROVED** [review]
lgtm :)
